src/lib/solana/index.ts

```diff
@@ -98,11 +172,14 @@ export class SolanaUtils {
    * Send SOL transfer transaction with memo
    */
   static async sendTransferWithMemo(
+    privyConnectedSolanaWallets: PrivyConnectedWalletsParams,
     params: TransferWithMemoParams,
   ): Promise<string | null> {
-    const provider = await this.getPhantomProvider();
+    const provider = await this.getSolanaProvider(privyConnectedSolanaWallets);
     if (!provider) {
-      throw new Error('Phantom wallet not found or connection rejected');
+      throw new Error(
+        'Phantom / Privy connected wallets not found or connection rejected',
+      );
```

Comment:
TODO: add a model which lets user navigate to their account page to connect a wallet via privy when this error occurs for a better ux

Reply:
<img width="731" alt="Screenshot 2025-03-31 at 12 51 51 AM" src="https://github.com/user-attachments/assets/683d0bcf-ee86-4fcd-aef9-3efb7189efd2" />
Done...

Reply:
<img width="702" alt="Screenshot 2025-04-02 at 11 58 57 PM" src="https://github.com/user-attachments/assets/465f69d6-cba9-4e26-bf41-edeb502ce1e9" />

Replace hard coded phantom wallet option with a connect wallet option which allows a user to connect to any solana wallets